include/zephyr/zephyr.h

```diff
@@ -9,4 +9,6 @@
 
 #include <zephyr/kernel.h>
 
+#warning "<zephyr/zephyr.h> is deprecated, include <zephyr/kernel.h> instead"
```

Comment:
```
#if !defined(CONFIG_LEGACY_INCLUDE_PATH)
#warning "<zephyr/zephyr.h> is deprecated, include <zephyr/kernel.h> instead"
#endif
```

Reply:
adopted this solution, `#pragma message` requires extra infrastructure upstream to prevent its usage. Also, if we want to introduce soft-deprecation I think we need to consider all kind of deprecations (e.g. functions, macros).